### DISCUSSION TITLE
`extra.socket.protocol` set to `''` (empty string) in `onClose` event listener, for closure that do not connect due to deprecated sub-protocol

### DISCUSSION THREAD

**Discussion by loucadufault** [enisdenjo/graphql-ws#562] [Q&A]
I have noticed for Server, with clients that attempt to connect with `Sec-WebSocket-Protocol` header set to the old `graphql-ws`, when logging the `extra.socket.protocol` received by the `onClose` listener, it is an empty string. However, logging the `req.headers['sec-websocket-protocol']` shows the populated value.

Is this something that is expected? I would prefer to not have to cover both cases in the handling logic.

**Comment by enisdenjo** [comment]
Hey hey, this is unrelated to graphql-ws itself. The `extra.socket` is [Node's `ws`](https://github.com/websockets/ws) socket instance (when using `graphql-ws/lib/use/ws`).

Briefly going through their code, the socket.protocol is set [here](https://github.com/websockets/ws/blob/ddfe4a804d79e7788ab136290e609f91cf68423f/lib/websocket.js#L963), could be that the connection fails earlier?
